easybuild/tools/filetools.py

```diff
@@ -1628,6 +1629,65 @@ def copy(paths, target_path, force_in_dry_run=False):
             raise EasyBuildError("Specified path to copy is not an existing file or directory: %s", path)
 
 
+def get_source_from_git(filename, targetdir, git_config):
+    """
+    Downloads a git repository, at a specific tag or commit, recursively or not, and make an archive with it
+
+    :param filename: name of the archive to save the code to (must be .tar.gz)
+    :param targetdir: target directory where to save the archive to
+    :param git_config: dictionary containing url, repo_name, recursive, and one of tag or commit
+    """
+    # if a non-empty dictionary was provided, download from git and archive
+    if not isinstance(git_config, dict):
+        raise EasyBuildError("Found a non null git_config in 'sources', but value is not a dictionary")
```

Comment:
@mboisson There's no link to `sources` here, `get_source_from_git` can also be used in other places in the future, so I would rephrase this to: `"Found unexpected type of value for 'git_config' argument: %s" % type(git_config)`

Reply:
fixing in next commit